### ISSUE TITLE
.getc won't permit completely raw input

### ISSUE THREAD

**Issue by geekosaur** [krunen/term-termios#10]
The example code is wrong in that the `getc` method will always wait for additional input in order to determine whether it is a combining character or not.

To actually do raw input, you need to set the `IO::Handle` (e.g. `$*IN`) to binary mode (`$*IN.encoding(Nil)`;) and use `$*IN.read(1)` to read a single byte; note that this will give you a `Buf` instead of a `Str`. (A quick hack for that is `$*IN.read(1).decode("latin-1")`, especially since `Buf` is currently not as easy to work with as `Str`.)

The raw input mechanism *may* require a somewhat recent `moarvm` and matching `rakudo` to work properly.

**Comment by AlexDaniel** [comment]
There's a rakudobug that I am rejecting with similar reasoning: [RT#125828](https://rt.perl.org/Ticket/Display.html?id=125828). I think something like ``INIT { $*IN.encoding: ‘ASCII’ }`` should work too.

**Comment by krunen** [comment]
I have made a module [Unicode::UTF8Parser](http://github.cim/krunen/unicode-utf8-parser) that parses UTF8 chars from a stream without waiting for combining chars. I can change the example to use this module.

I have held back doing this, as I think Rakudo should expose this as some kind of parser mode. It seems silly to reimplement a full UTF8 parser in plain perl6. Also, I am not sure how Rakudo will handle un-normalized unicode (that is, with base char and combining chars separately). I have used it myself without much problems though - the strings seems to become normalized quite quickly.
